app/controllers/api/v2/mail_notifications_controller.rb

```diff
@@ -21,6 +22,54 @@ def index
 
       def show
       end
+
+      api :POST, "/users/:user_id/mail_notifications", N_("Create an email notification for a user")
+      param :user_id, String, :required => true
+      param :id, Integer, :required => true
+      param :interval, String, :required => true
+      param :mail_query, String, :required => false, :desc => N_("Relevant only for audit summary notification")
+
+      def create
+        @user = User.find(params[:user_id])
+        mail_notification = MailNotification.find(params[:id])
+        interval_values = mail_notification.subscription_type == 'alert' ? mail_notification.subscription_options : MailNotification::INTERVALS
+        raise ::Foreman::Exception.new(N_("Interval option is not valid")) unless interval_values.include? params[:interval].capitalize
+        raise ::Foreman::Exception.new(N_("Mail query is not valid")) unless params[:mail_query].nil? || mail_notification.queryable?
+        user_mail_notification = @user.user_mail_notifications.find { |mail_notification| mail_notification.mail_notification_id == params[:id] }
+        if user_mail_notification.nil?
+          mail_notifications_params = {'interval' => params[:interval].capitalize,
+                                       'mail_notification_id' => params[:id],
+                                       'user_id' => params[:user_id],
+                                       'mail_query' => params[:mail_query]}
+          @user.user_mail_notifications << UserMailNotification.new(mail_notifications_params)
+        else
+          user_mail_notification[:interval] = params[:interval].capitalize
+          user_mail_notification[:mail_query] = params[:mail_query]
+          @user.user_mail_notifications << user_mail_notification
+        end
+      end
```

Comment:
If I'm not mistaken, for each create action we provide something as a response, e.g. object that was created.

Reply:
I fixed it, it is now returning the user_mail_notification that is created or modified.